### DISCUSSION TITLE
Does version 8.2.6 train classification better than the latest YOLOv11 version?

### DISCUSSION THREAD

**Discussion by wuyou-zhang** [ultralytics/ultralytics#20154] [Q&A]
Training the MNIST160 classification dataset has shown a decline in performance starting from version 8.2.8. Version 8.2.7 fails during training, and the latest YOLOv11 version's classification performance is also inferior to that of version 8.2.6.Object detection still needs to be tested.
model = YOLO(r'yolov8x-cls.pt')
if __name__ == '__main__':
     results = model.train(data='./mnist160', epochs=30, imgsz=64,resume=False)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @wuyou-zhang, thank you for sharing your observations and for your interest in Ultralytics 🚀! We appreciate your detailed feedback regarding classification training performance across versions.

If this is a 🐛 Bug Report, could you please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate further? Include any relevant logs, error messages, or additional details about your setup (e.g., OS, Python version, and GPU/CPU environment).

If this is a performance-related ❓ Question, it would be helpful to share more information, such as:
- Your `data.yaml` configuration for the MNIST160 dataset.
- Training logs or metrics from the versions you tested.
- Any modifications made to the training process.

In the meantime, please ensure you are using the latest `ultralytics` package and dependencies. Upgrade your environment with:

```bash
pip install -U ultralytics
```

You can also explore the following verified environments for running YOLO to ensure consistency:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

For further insights, you can check out the [Docs](https://docs.ultralytics.com/) for guidance on [model training tips](https://docs.ultralytics.com/guides/model-training-tips/) and [classification tasks](https://docs.ultralytics.com/tasks/classification/).

Lastly, join the Ultralytics community for additional support:
- Real-time chat on [Discord](https://discord.com/invite/ultralytics) 🎧
- In-depth discussions on [Discourse](https://community.ultralytics.com/)
- Knowledge sharing on our [Subreddit](https://reddit.com/r/Ultralytics)

This is an automated response to assist you quickly, but an Ultralytics engineer will review your issue and provide further guidance soon. Thank you for your patience and for contributing to the Ultralytics community! 😊

**Reply by glenn-jocher** [reply]
Hi @wuyou-zhang, thanks for reporting this potential classification performance regression. We'd like to investigate this issue further to determine if there's a genuine regression across versions.

Could you share more specific metrics that show the performance decline? For example, accuracy values from version 8.2.6 compared to YOLO11 when using identical training parameters. Also, for the 8.2.7 training failure, any error messages would be helpful.

In the meantime, you might want to file a specific issue on our GitHub repository with these details so our engineering team can thoroughly investigate and potentially fix the regression in future releases. If you need to use version 8.2.6 for critical projects until this is resolved, that's a reasonable approach.

We appreciate your feedback as it helps us improve our models and maintain performance across different tasks.

**Reply by wuyou-zhang** [reply]
8.2.6and yolo11
![屏幕截图 2025-04-14 144718](https://github.com/user-attachments/assets/ffa3dff3-0562-4e9b-88fb-c63b4af64c87)
![屏幕截图 2025-04-14 144734](https://github.com/user-attachments/assets/6f8d15ae-f6be-48fe-8de6-0fc605da568b)

**Reply by glenn-jocher** [reply]
Thanks for sharing those comparison screenshots @wuyou-zhang. I can see the performance difference you're experiencing between 8.2.6 and YOLO11 on the MNIST160 classification task. This appears to be a legitimate regression that warrants investigation.

Could you please create a formal issue in our [Issues tracker](https://github.com/ultralytics/ultralytics/issues) with:
1. These screenshots
2. Your exact training command/code
3. The dataset structure details
4. Environment information (OS, Python version, GPU)

In the meantime, for critical projects requiring classification, continuing with 8.2.6 is a reasonable workaround. We'll prioritize investigating this regression and work to restore or improve classification performance in upcoming releases.
